scripts/build_docs.py

```diff
@@ -0,0 +1,130 @@
+    current version and see if it has changed (which would mean the docs need re-building).
+    """
+    clean_filename = file_name.replace(".py", "")
+    markdown_filename = f"{clean_filename}.md"
+
+    output_path = os.path.join(docs_dir, relative_path, markdown_filename)
+    nav_path = os.path.join("api", relative_path, markdown_filename)
+    if not check:
+        render_docs(relative_path, file_name, output_path)
+
+    return {os.path.basename(clean_filename): nav_path}
+
+
+def build_docs(root_path: str, docs_dir: str, check: bool = False):
+
+    nav_root = []
+
+    for child in os.listdir(root_path):
+        relative_path = os.path.join(root_path, child)
+
+        if (
+            "__pycache__" in relative_path
+            or "tests" in relative_path
+            or "mypy_cache" in relative_path
+        ):
+            continue
+
+        without_allennlp = str(root_path).replace("allennlp/", "")
+        target_dir = os.path.join(docs_dir, without_allennlp)
+        if not os.path.exists(target_dir):
+            os.mkdir(target_dir)
+
+        if os.path.isdir(relative_path):
+            nav_subsection = build_docs(relative_path, docs_dir, check)
+            if not nav_subsection:
+                continue
+            nav_root.append({child: nav_subsection})
+
+        else:
+            if child in exclude_files or not child.endswith(".py"):
+                continue
+
+            nav = build_docs_for_file(without_allennlp, child, docs_dir, check)
+            nav_root.append(nav)
+
+    return nav_root
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Builds the Markdown docs for allennlp.")
+    parser.add_argument(
+        "--check", help="If passed, just check if the docs need rebuilding.", action="store_true"
+    )
+
+    args = parser.parse_args()
+    if args.check:
+        print("Checking if changes are required to the auto-generated documentation.")
+    else:
+        print("Building the docs.")
+
+    parent_folder_path = Path(__file__).parent.parent
+    yaml_path = parent_folder_path / "mkdocs.yml"
+    source_path = parent_folder_path / "allennlp"
+    docs_dir = str(parent_folder_path / "docs" / "api")
+    yaml = YAML()
+
+    nav_entries = build_docs(source_path, docs_dir, check=args.check)
+    nav_entries.sort(key=lambda x: list(x)[0], reverse=False)
+
+    mkdocs_yaml = yaml.load(yaml_path)
+    docs_key = "API"
+    site_nav = mkdocs_yaml["nav"]
+
+    # Find the yaml corresponding to the API
+    for nav_obj in site_nav:
+        if docs_key in nav_obj:
+            break
+
+    if args.check:
+        if nav_obj[docs_key] != nav_entries:
+            raise ValueError(
+                "Changes are required to the API documentation. Run `./scripts/build_docs.py` and commit the changes."
+            )
+        else:
+            print("No changes required!")
```

Comment:
Yeah, looking at this whole `check` logic, I don't think it does what we want.  We'd want to update minor docstring fixes, and those wouldn't be caught.  We should probably just remove the whole `check` stuff, and always rebuild on master commits.

Reply:
We could also potentially run `render_file` on all files that were touched in a specific PR, to make sure there aren't rendering errors introduced by that PR.  But that doesn't need to be done now.